vault/resource_aws_auth_backend_identity_whitelist.go

```diff
@@ -0,0 +1,153 @@
+package vault
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/vault/api"
+)
+
+var (
+	awsAuthBackendIdentityWhitelistBackendFromPathRegex = regexp.MustCompile("^auth/(.+)/config/tidy/identity-whitelist$")
+)
+
+func awsAuthBackendIdentityWhitelistResource() *schema.Resource {
+	return &schema.Resource{
+		Create: awsAuthBackendIdentityWhitelistWrite,
+		Read:   awsAuthBackendIdentityWhitelistRead,
+		Update: awsAuthBackendIdentityWhitelistWrite,
+		Delete: awsAuthBackendIdentityWhitelistDelete,
+		Exists: awsAuthBackendIdentityWhitelistExists,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"backend": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Unique name of the auth backend to configure.",
+				ForceNew:    true,
+				Default:     "aws",
+				// standardise on no beginning or trailing slashes
+				StateFunc: func(v interface{}) string {
+					return strings.Trim(v.(string), "/")
+				},
+			},
+			"safety_buffer": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "The amount of extra time that must have passed beyond the roletag expiration, before it's removed from backend storage.",
+			},
+			"disable_periodic_tidy": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "If true, disables the periodic tidying of the identiy whitelist entries.",
+			},
+		},
+	}
+}
+
+func awsAuthBackendIdentityWhitelistWrite(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	backend := d.Get("backend").(string)
+	data := map[string]interface{}{}
+
+	if v, ok := d.GetOk("safety_buffer"); ok {
+		data["safety_buffer"] = v
+	}
+	if v, ok := d.GetOk("disable_periodic_tidy"); ok {
```

Comment:
`disable_periodic_tidy` needs to have a default value, or use `d.GetOkExists`. Without one of those, a user cannot set this to `true` and then remove it, or set it to `false`.

Reply:
Great catch, thanks. Updated.